embassy-stm32f4/src/can.rs

```diff
@@ -0,0 +1,80 @@
+//! Async low power Serial.
+//!
+//! The peripheral is autmatically enabled and disabled as required to save power.
+//! Lowest power consumption can only be guaranteed if the send receive futures
+//! are dropped correctly (e.g. not using `mem::forget()`).
+
+use bxcan;
+use bxcan::Interrupts;
+use core::future::Future;
+use embassy::interrupt::Interrupt;
+use embassy::util::InterruptFuture;
+use embassy::util::Signal;
+use nb;
+use nb::block;
+
+use crate::hal::prelude::*;
+use crate::interrupt;
+
+/// Interface to the Serial peripheral
+pub struct Can<T: bxcan::Instance> {
+    can: bxcan::Can<T>,
+    tx_int: interrupt::CAN1_TX,
```

Comment:
the interrupts may be wrong if user doesn't use CAN1? 

How does `Instance` work in bxcan? the user has to implement it themselves for their chip?  Maybe we can provide the impls?